layers/+lang/factor/funcs.el

```diff
@@ -65,21 +65,31 @@ Will stop current fuel connection if applicable."
     (require 'fuel-mode)
     (require 'factor-mode)))
 
+(defvar factor--previous-fuel-path nil)
+
 (defun factor//reload-fuel-from-path (path)
   "Unload current emacs lisp fuel implementation and load the on from PATH.
 
 Since unloading switches buffers which were in factor-mode back
 to fundamental mode, this re-enables factor-mode in these buffers
-afterwards."
-  (let ((factor-buffers (cl-loop for b being the buffers
-                                 if (eq (buffer-local-value 'major-mode b)
-                                        'factor-mode)
-                                 collect b)))
-    (factor//unload-fuel)
-    (factor//load-fuel-from-path path)
-    (cl-loop for b in factor-buffers do
-             (with-current-buffer b
-               (factor-mode)))))
+afterwards.
+
+Only reloads if value of `factor--previous-fuel-path' differs from PATH.
+
+"
+  (when (and factor--previous-fuel-path
+             (not (eql factor--previous-fuel-path path)))
+    (let ((factor-buffers (cl-loop for b being the buffers
+                                   if (eq (buffer-local-value 'major-mode b)
+                                          'factor-mode)
+                                   collect b)))
+      (factor//unload-fuel)
+      (message "Reloading fuel mode from %s" path)
+      (factor//load-fuel-from-path path)
+      (cl-loop for b in factor-buffers do
+               (with-current-buffer b
+                 (factor-mode)))))
+  (setq factor--previous-fuel-path path))
```

Comment:
```elisp
(defun factor//reload-fuel-from-path (path)
  (when (and factor--previous-fule-path
             (not (eql factor-previous-fule-path path)))
    (factor//unload-fuel)
    (message "Reloading fuel mode from %s" path)
    (factor//load-fuel-from-path path)
    (cl-loop for b in (buffer-list)
             if (eq (buffer-local-value 'major-mode b)
                    'factor-mode)
             do (with-current-buffer b
                  (factor-mode +1))))
  (setq factor--previous-fuel-path path))
```

This is shorter. Also `cl-loop for FOO in BAR` is slightly faster than `cl-loop for FOO being the BAR`.

Reply:
Thanks. Will change that.

Reply:
Actually, that modification does not seem to work.  I need to collect the list of factor-mode buffers before unloading, because `major-mode` will not be `factor-mode` anymore after unloading.

Reply:
> ```emacs-lisp
> (defun factor//reload-fuel-from-path (path)
>   (when (and factor--previous-fule-path
>              (not (eql factor-previous-fule-path path)))
>     (factor//unload-fuel)
>     (message "Reloading fuel mode from %s" path)
>     (factor//load-fuel-from-path path)
>     (cl-loop for b in (buffer-list)
>              if (eq (buffer-local-value 'major-mode b)
>                     'factor-mode)
>              do (with-current-buffer b
>                   (factor-mode +1))))
>   (setq factor--previous-fuel-path path))
> ```
> 
> This is shorter. Also `cl-loop for FOO in BAR` is slightly faster than `cl-loop for FOO being the BAR`.

Reply:
As the loop contents has not been changed this can already be merged. Optimisation of the loop should be done in a separate PR.